bin/broadcast_prs.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+
+from cookiecutter.main import cookiecutter
+import json
+import os
+import requests
+from requests.auth import HTTPBasicAuth
+import sys
+import subprocess
+
+# The GitHub base url or the nf-core project
+GH_BASE_URL = "https://github.com/nf-core"
+# The current cookiecutter template url for nf-core pipelines
+NF_CORE_TEMPLATE = "https://github.com/nf-core/cookiecutter.git"
+# The JSON file is updated on every push event on the nf-core GitHub project
+NF_CORE_PIPELINE_INFO = "http://nf-co.re/pipelines.json"
+# The API endpoint for creating pull requests
+GITHUB_PR_URL_TEMPL = "https://api.github.com/repos/nf-core/{pipeline}/pulls"
+
+
+class UpdateTemplate:
+    """Updates the template content of an nf-core pipeline in
+    its `TEMPLATE` branch.
+
+    Args: - pipeline: The pipeline name
+          - context: a parsed dictionary of a cookiecutter.json file
+          - branch: The template branch name, default=`TEMPLATE`
+    """
+
+    def __init__(self, pipeline, context, branch='TEMPLATE'):
+        """Basic constructor
+        """
+        self.pipeline = pipeline
+        self.repo_url = "{base_url}/{pipeline}".format(
+                    base_url=GH_BASE_URL,
+                    pipeline=pipeline)
+        self.context = context
+        self.branch = branch
+    
+    def run(self):
+        """Execute the template update.
+        """
+        self._clone_repo()
+        self._apply_changes()
+        self._commit_changes()
+    
+    def _clone_repo(self):
+        """Clone the repo and switch to the configured branch.
+        """
+        subprocess.run(["git", "clone", self.repo_url, "-b", self.branch, self.pipeline])
+
+    def _apply_changes(self):
+        """Apply the changes of the cookiecutter template
+        to the pipelines template branch.
+        """
+        cookiecutter(NF_CORE_TEMPLATE,
```

Comment:
Instead of using cookiecutter, you can now use `nf-core create`:

```python
from nf_core.create import run_cookiecutter
run_cookiecutter(self.pipeline_name, self.pipeline_description, self.pipeline_version)
```

Reply:
NB: I should probably add an option to specify an output directory. At the moment it just spits it out into the current working directory I think. I guess you'll want to use `self.tmpdir`.

Also note that currently it creates the files in a subdirectory called self.pipeline_name. I haven't figured an easy way to avoid this, though I guess the `create` command could also use a temp directory and then copy.